### PR TITLE
feat: Qdrant external retriever

### PR DESCRIPTION
**Pull request by Anush008**
## Description

This PR adds support for external retrieval using Qdrant - https://qdrant.tech. Includes docs and examples.

Unit and E2E tests using a local Qdrant instance have been set up.

## Type of Change

- [x] New feature
- [x] Documentation update

## Note

I have agreed to and signed the CLA(cla@neo4j.com).

### CODE REVIEW

**Comment by Anush008** [comment]
Could someone please run
```
poetry lock --no-update 
```
on this branch and commit?
It's been over 3500 seconds since mine began.

File: pyproject.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ from = "src"
 python = "^3.9.0"
 neo4j = "^5.17.0"
 pydantic = "^2.6.3"
+urllib3 = "<2"
```
Comment (Anush008) [inline pyproject.toml:34]: Without this,
```bash
poetry lock
```
takes forever.

File: docs/source/api.rst [hunk 2]
```diff
@@ -147,6 +147,12 @@ PineconeNeo4jRetriever
 .. autoclass:: neo4j_graphrag.retrievers.external.pinecone.pinecone.PineconeNeo4jRetriever
     :members: search
 
+QdrantNeo4jRetriever
+======================
```
Comment (willtai) [inline docs/source/api.rst:151]: [Best practice](https://www.pgdp.net/wiki/RST:_Best_Practices#:~:text=example%20is%20optional.-,Chapters%20and%20Sections,-Chapters%20should%20be) for this to have the same length as the header

```suggestion
====================
```

File: docs/source/user_guide_rag.rst [hunk 3]
```diff
@@ -672,6 +674,35 @@ Pinecone Retrievers
 
 Also see :ref:`pineconeneo4jretriever`.
 
+.. _qdrant-neo4j-retriever-user-guide:
+
+Qdrant Retrievers
+-------------------
```
Comment (willtai) [inline docs/source/user_guide_rag.rst:680]:
```suggestion
-----------------
```

Comment (Anush008) [reply docs/source/user_guide_rag.rst]: 👍

**Review by stellasia — APPROVED** [review]
Thank you for your patience, nice work!

**Pull request merged**